eldritch/encounters.py

```diff
@@ -144,11 +144,101 @@ def Society4(char):
   cond = events.Conditional(char, check, "successes", {0: events.Nothing(), 2: skill})
   return events.Sequence([check, cond], char)
 
+def Administration1(char):
+  check = events.Check(char, "lore", -1)
+  dollars = events.Gain(char, {"dollars": 5})
+  return events.PassFail(char, check, dollars, events.Nothing())
+def Administration2(char):
+  return events.Gain(char, {"clues": 1})
+def Administration3(char):
+  check =  events.Check(char, "will", -1)
+  retainer = events.StatusChange(char, "retainer")
+  return events.PassFail(char, check, retainer, events.Nothing())
+def Administration4(char):
+  check = events.Check(char, "lore", -2)
+  spell = events.Draw(char, "spells", 1)
+  curse = events.Curse(char)
+  return events.PassFail(char, check, spell, curse)
+def Administration5(char):
+  return events.ForceMovement(char, "Asylum") # TODO: also have an event
+def Administration6(char):
+  # Administration 6 is identical to Administration 1
+  return Administration1(char)
 def Administration7(char):
   check = events.Check(char, "will", -2)
   gain = events.Gain(char, {"dollars": 8})
   arrest = events.Arrested(char)
   return events.PassFail(char, check, gain, arrest)
+
+def Library1(char):
+  check = events.Check(char, "will", -1)
+  tome = events.Draw(char, "unique", 1) # TODO: this is actually draw the first tome
+  move = events.ForceMovement(char, "University")
+  return events.PassFail(char, check, tome, move)
+def Library2(char):
+  check = events.Check(char, "will", 0)
+  move = events.ForceMovement(char, "University")
+  spell = events.Draw(char, "spells", 2)
+  unique = events.Draw(char, "unique", 1)
+  cond = events.Conditional(char, check, "successes", {0: move, 1: spell, 2: unique})
+  return events.Sequence([check, cond], char)
+def Library3(char):
+  check = events.Check(char, "lore", -2)
+  die = events.DiceRoll(char, 1)
+  gain = events.Gain(char, {"clues": die})
+  loss = events.Loss(char, {"stamina": 2, "sanity": 2})
+  return events.PassFail(char, check, events.Sequence([die, gain]), loss)
```

Comment:
You should add the character to the Sequence here.

Reply:
hmm, surprised that it worked without that.

Reply:
The sequence can work without a character attached to it. However, if the character happens to go insane when resolving some event, then the events on the stack that will be unwound are only the events that are attached to that specific character. In this case, it's hard to see how gaining clue tokens would make the character go insane, but it's good to keep in mind for future changes.